accuracy_metric.py

```diff
@@ -0,0 +1,161 @@
+    cor_words, inc_words = get_words(cor_line), get_words(inc_line)
+    trg_words = get_words(trg_line)
+    trg_correct, trg_incorrect = get_trg_correct_incorrect(cor_words, inc_words, trg_words)
+
+    if trg_incorrect:
+        decision = 'Incorrect'
+    else:
+        decision = 'Correct'
+
+    return [decision, trg_correct, trg_incorrect]
+
+
+def read_file_to_list(filename):
+    """
+    Read a file to a list of lines; each line will be lowercased and punctuation will be removed.
+
+    :param filename: File to read.
+    :return: List of lines of the file.
+    """
+    out_list = []
+    with open(filename, 'r') as infile:
+        for line in infile:
+            cleaned_line = clean_line(line)
+            out_list.append(cleaned_line)
+    return out_list
+
+
+def accuracy_metric(hypothesis, cor_ref, inc_ref):
+    """
+    Compute whether gender is correct in the translation given actual reference and counterfactual reference
+    We assume input files contain one sentence per line
+    :param hypothesis: AMT hypothesis file to be evaluated
+    :param cor_ref: Correctly gendered reference file path
+    :param inc_ref: Incorrect (counterfactual) reference file path
+    :return decision (Corrct/Incorrect), one per each line in the input files
+    """
+    # read in the files. 
+    # Note that each line of the input files will be lowercased and punctuation will be removed. 
+    trg_list = read_file_to_list(hypothesis)
+    cor_list = read_file_to_list(cor_ref)
+    inc_list = read_file_to_list(inc_ref)
+
+    assert len(trg_list) == len(cor_list), f'Output file and original reference file must have the same number of lines. Files are {hypothesis}, {cor_ref}'
+    assert len(trg_list) == len(inc_list), f'Output file and counterfactual reference file must have the same number of lines. Files are {hypothesis},  {inc_ref}'
+
+    metric_annot_mapped = []    
+    for trg_line, cor_line, inc_line in zip(trg_list, cor_list, inc_list):
+        [decision, trg_correct, trg_incorrect] = gender_decision(trg_line, cor_line, inc_line)
+        metric_annot_mapped.append(decision) 
+    accuracy = metric_annot_mapped.count('Correct')/len(metric_annot_mapped)
+    return accuracy, metric_annot_mapped
+
+
+def logicaloperation_AND(decision1, decision2):
+    """
+    Performs AND operation (per line) on the inputs: returns 'Correct' only if both inputs are 'Correct'. 
+    :param decision1 and decision2: lists with each line either 'Correct' or 'Incorrect'
+    :return: accuracy and list of decisions
+    """
+    combined_decision = ['Incorrect' if 'Incorrect' in [d1,d2] else 'Correct' for d1,d2 in zip(decision1, decision2)]
+    accuracy = combined_decision.count('Correct')/len(combined_decision)
+    return accuracy, combined_decision
+
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--target_lang', type=str, help='one of ar,de,es,fr,hi,it,pt,ru')
```

Comment:
It would be nice here to set a variable containing the list of allowed target language abbreviations. Then you can both call the list in the help string, and also check against it when reading in the arguments.